executor/executor.go

```diff
@@ -370,14 +372,60 @@ func (e *LimitExec) Next(goCtx goctx.Context) (Row, error) {
 	if srcRow == nil {
 		return nil, nil
 	}
-	e.Idx++
+	e.cursor++
 	return srcRow, nil
 }
 
+// NextChunk implements the Executor NextChunk interface.
+func (e *LimitExec) NextChunk(chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.cursor >= e.end {
+		return nil
+	}
+	for !e.meetFirstBatch {
+		err := e.children[0].NextChunk(e.childrenResults[0])
+		if err != nil {
+			return errors.Trace(err)
+		}
+		batchSize := uint64(e.childrenResults[0].NumRows())
+		// no more data.
+		if batchSize == 0 {
+			return nil
+		}
+		e.cursor += batchSize
```

Comment:
I think to update the cursor after we append the data is easier to understand.
Then the cursor will never be greater than e.end.

Reply:
It's not that hard to understand and if we update cursor after appending data the code will be more or less ugly.

Reply:
I think there is a way to make the code easier to understand.